core/tools.js

```diff
@@ -1627,6 +1627,21 @@
 					yellowgreen: '#9ACD32'
 				},
 
+				_borderStyle: [
+					'none',
+					'hidden',
+					'dotted',
+					'dashed',
+					'solid',
+					'double',
+					'groove',
+					'ridge',
+					'inset',
+					'outset'
+				],
+
+				_widthRegExp: /^(thin|medium|thick|[\+-]?\d+(\.?\d+|)[a-z%]+|[\+-]?0+(\.?0+|)|\.\d+[a-z%]+)$/,
```

Comment:
RegExp works fine. I was just wondering about `(\.?\d+|)` and `(\.?0+|)`. Isn't it an equivalent to `(\.\d+)?` and `(\.0+)?` in this case? Which version do you find more readable?